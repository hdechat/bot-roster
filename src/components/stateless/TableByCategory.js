import React from 'react';
import Robot from './Robot';

function TableByCategory({ header, robots, updateName, deleteRobot, inRosters }) {
  const team = robots.map(bot => 
    <Robot 
      key={bot.firstName + bot.lastName + bot.totalAttrScore}
      robot={bot}
      updateName={updateName}
      deleteRobot={deleteRobot}
      inRosters={inRosters} />
  );

  return(
    <table>
      <thead>
        <tr className="header__titles">
          <th>ID</th>
          <th>FIRST NAME</th>
          <th>LAST NAME</th>
          <th>SPEED</th>
          <th>STRENGTH</th>
          <th>AGILITY</th>
          <th>TTL ATTRIBUTES SCORE</th>
        </tr>
      </thead>
      <thead>
        <tr>
          <th className="header__category" colSpan="7">{ header }</th>
        </tr>
      </thead>
      <tbody>
        { team }
      </tbody>
    </table>
  )
}

export default TableByCategory;
