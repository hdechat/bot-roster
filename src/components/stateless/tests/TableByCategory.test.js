import React from 'react';
import TableByCategory from '../TableByCategory';
import { shallow } from 'enzyme';

describe('TableByCategory', () => {
  let wrapper;
  let mockHeader;
  let mockRobots;

  beforeEach(() => {
    mockHeader = "Starters";
    mockRobots = [
      {
        id: 'ABC123',
        firstName: 'Twikki',
        lastName: 'Rogers',
        speed: 20,
        strength: 30,
        agility: 40,
        totalAttrScore: 90
      },
      {
        id: 'ZXY987',
        firstName: 'R2',
        lastName: 'D2',
        speed: 5,
        strength: 8,
        agility: 9,
        totalAttrScore: 22
      }
    ];

    wrapper = shallow(
      <TableByCategory header={mockHeader} robots={mockRobots} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders all the expected elements', () => {
    expect(wrapper.find('table').length).toEqual(1);
    expect(wrapper.find('thead').length).toEqual(1);
    expect(wrapper.find('tr').length).toEqual(1);
    expect(wrapper.find('th').length).toEqual(1);
    expect(wrapper.find('tbody').length).toEqual(1);
  });

  it('renders the header prop value within the th element', () => {
    expect(wrapper.find('th').text()).toBe(mockHeader)
  });

  it('renders the robot objects', () => {
    expect(wrapper.find('Robot').length).toEqual(2);
  });
});
