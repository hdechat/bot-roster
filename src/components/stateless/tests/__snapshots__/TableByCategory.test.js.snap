// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableByCategory matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableByCategory
    header="Starters"
    robots={
        Array [
            Object {
              "agility": 40,
              "firstName": "Twikki",
              "id": "ABC123",
              "lastName": "Rogers",
              "speed": 20,
              "strength": 30,
              "totalAttrScore": 90,
            },
            Object {
              "agility": 9,
              "firstName": "R2",
              "id": "ZXY987",
              "lastName": "D2",
              "speed": 5,
              "strength": 8,
              "totalAttrScore": 22,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
                    <th
                              className="header__category"
                              colSpan="7"
                    >
                              Starters
                    </th>
          </tr>
</thead>,
        <thead>
          <tr
                    className="header__titles"
          >
                    <th>
                              ID
                    </th>
                    <th>
                              FIRST NAME
                    </th>
                    <th>
                              LAST NAME
                    </th>
                    <th>
                              SPEED
                    </th>
                    <th>
                              STRENGTH
                    </th>
                    <th>
                              AGILITY
                    </th>
                    <th>
                              TTL ATTRIBUTES SCORE
                    </th>
          </tr>
</thead>,
        <tbody>
          <Robot
                    deleteRobot={undefined}
                    inRosters={undefined}
                    robot={
                              Object {
                                        "agility": 40,
                                        "firstName": "Twikki",
                                        "id": "ABC123",
                                        "lastName": "Rogers",
                                        "speed": 20,
                                        "strength": 30,
                                        "totalAttrScore": 90,
                                      }
                    }
                    updateName={undefined}
          />
          <Robot
                    deleteRobot={undefined}
                    inRosters={undefined}
                    robot={
                              Object {
                                        "agility": 9,
                                        "firstName": "R2",
                                        "id": "ZXY987",
                                        "lastName": "D2",
                                        "speed": 5,
                                        "strength": 8,
                                        "totalAttrScore": 22,
                                      }
                    }
                    updateName={undefined}
          />
</tbody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th
                        className="header__category"
                        colSpan="7"
            >
                        Starters
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <th
              className="header__category"
              colSpan="7"
>
              Starters
</th>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Starters",
              "className": "header__category",
              "colSpan": "7",
            },
            "ref": null,
            "rendered": "Starters",
            "type": "th",
          },
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr
            className="header__titles"
>
            <th>
                        ID
            </th>
            <th>
                        FIRST NAME
            </th>
            <th>
                        LAST NAME
            </th>
            <th>
                        SPEED
            </th>
            <th>
                        STRENGTH
            </th>
            <th>
                        AGILITY
            </th>
            <th>
                        TTL ATTRIBUTES SCORE
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                ID
</th>,
              <th>
                FIRST NAME
</th>,
              <th>
                LAST NAME
</th>,
              <th>
                SPEED
</th>,
              <th>
                STRENGTH
</th>,
              <th>
                AGILITY
</th>,
              <th>
                TTL ATTRIBUTES SCORE
</th>,
            ],
            "className": "header__titles",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ID",
              },
              "ref": null,
              "rendered": "ID",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "FIRST NAME",
              },
              "ref": null,
              "rendered": "FIRST NAME",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LAST NAME",
              },
              "ref": null,
              "rendered": "LAST NAME",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SPEED",
              },
              "ref": null,
              "rendered": "SPEED",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "STRENGTH",
              },
              "ref": null,
              "rendered": "STRENGTH",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AGILITY",
              },
              "ref": null,
              "rendered": "AGILITY",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TTL ATTRIBUTES SCORE",
              },
              "ref": null,
              "rendered": "TTL ATTRIBUTES SCORE",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Robot
              deleteRobot={undefined}
              inRosters={undefined}
              robot={
                            Object {
                                          "agility": 40,
                                          "firstName": "Twikki",
                                          "id": "ABC123",
                                          "lastName": "Rogers",
                                          "speed": 20,
                                          "strength": 30,
                                          "totalAttrScore": 90,
                                        }
              }
              updateName={undefined}
/>,
            <Robot
              deleteRobot={undefined}
              inRosters={undefined}
              robot={
                            Object {
                                          "agility": 9,
                                          "firstName": "R2",
                                          "id": "ZXY987",
                                          "lastName": "D2",
                                          "speed": 5,
                                          "strength": 8,
                                          "totalAttrScore": 22,
                                        }
              }
              updateName={undefined}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "TwikkiRogers90",
            "nodeType": "function",
            "props": Object {
              "deleteRobot": undefined,
              "inRosters": undefined,
              "robot": Object {
                "agility": 40,
                "firstName": "Twikki",
                "id": "ABC123",
                "lastName": "Rogers",
                "speed": 20,
                "strength": 30,
                "totalAttrScore": 90,
              },
              "updateName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "R2D222",
            "nodeType": "function",
            "props": Object {
              "deleteRobot": undefined,
              "inRosters": undefined,
              "robot": Object {
                "agility": 9,
                "firstName": "R2",
                "id": "ZXY987",
                "lastName": "D2",
                "speed": 5,
                "strength": 8,
                "totalAttrScore": 22,
              },
              "updateName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th
                                    className="header__category"
                                    colSpan="7"
                        >
                                    Starters
                        </th>
            </tr>
</thead>,
          <thead>
            <tr
                        className="header__titles"
            >
                        <th>
                                    ID
                        </th>
                        <th>
                                    FIRST NAME
                        </th>
                        <th>
                                    LAST NAME
                        </th>
                        <th>
                                    SPEED
                        </th>
                        <th>
                                    STRENGTH
                        </th>
                        <th>
                                    AGILITY
                        </th>
                        <th>
                                    TTL ATTRIBUTES SCORE
                        </th>
            </tr>
</thead>,
          <tbody>
            <Robot
                        deleteRobot={undefined}
                        inRosters={undefined}
                        robot={
                                    Object {
                                                "agility": 40,
                                                "firstName": "Twikki",
                                                "id": "ABC123",
                                                "lastName": "Rogers",
                                                "speed": 20,
                                                "strength": 30,
                                                "totalAttrScore": 90,
                                              }
                        }
                        updateName={undefined}
            />
            <Robot
                        deleteRobot={undefined}
                        inRosters={undefined}
                        robot={
                                    Object {
                                                "agility": 9,
                                                "firstName": "R2",
                                                "id": "ZXY987",
                                                "lastName": "D2",
                                                "speed": 5,
                                                "strength": 8,
                                                "totalAttrScore": 22,
                                              }
                        }
                        updateName={undefined}
            />
</tbody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th
                            className="header__category"
                            colSpan="7"
              >
                            Starters
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <th
                className="header__category"
                colSpan="7"
>
                Starters
</th>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Starters",
                "className": "header__category",
                "colSpan": "7",
              },
              "ref": null,
              "rendered": "Starters",
              "type": "th",
            },
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr
              className="header__titles"
>
              <th>
                            ID
              </th>
              <th>
                            FIRST NAME
              </th>
              <th>
                            LAST NAME
              </th>
              <th>
                            SPEED
              </th>
              <th>
                            STRENGTH
              </th>
              <th>
                            AGILITY
              </th>
              <th>
                            TTL ATTRIBUTES SCORE
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  ID
</th>,
                <th>
                  FIRST NAME
</th>,
                <th>
                  LAST NAME
</th>,
                <th>
                  SPEED
</th>,
                <th>
                  STRENGTH
</th>,
                <th>
                  AGILITY
</th>,
                <th>
                  TTL ATTRIBUTES SCORE
</th>,
              ],
              "className": "header__titles",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ID",
                },
                "ref": null,
                "rendered": "ID",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "FIRST NAME",
                },
                "ref": null,
                "rendered": "FIRST NAME",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LAST NAME",
                },
                "ref": null,
                "rendered": "LAST NAME",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SPEED",
                },
                "ref": null,
                "rendered": "SPEED",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "STRENGTH",
                },
                "ref": null,
                "rendered": "STRENGTH",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AGILITY",
                },
                "ref": null,
                "rendered": "AGILITY",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TTL ATTRIBUTES SCORE",
                },
                "ref": null,
                "rendered": "TTL ATTRIBUTES SCORE",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Robot
                deleteRobot={undefined}
                inRosters={undefined}
                robot={
                                Object {
                                                "agility": 40,
                                                "firstName": "Twikki",
                                                "id": "ABC123",
                                                "lastName": "Rogers",
                                                "speed": 20,
                                                "strength": 30,
                                                "totalAttrScore": 90,
                                              }
                }
                updateName={undefined}
/>,
              <Robot
                deleteRobot={undefined}
                inRosters={undefined}
                robot={
                                Object {
                                                "agility": 9,
                                                "firstName": "R2",
                                                "id": "ZXY987",
                                                "lastName": "D2",
                                                "speed": 5,
                                                "strength": 8,
                                                "totalAttrScore": 22,
                                              }
                }
                updateName={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "TwikkiRogers90",
              "nodeType": "function",
              "props": Object {
                "deleteRobot": undefined,
                "inRosters": undefined,
                "robot": Object {
                  "agility": 40,
                  "firstName": "Twikki",
                  "id": "ABC123",
                  "lastName": "Rogers",
                  "speed": 20,
                  "strength": 30,
                  "totalAttrScore": 90,
                },
                "updateName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "R2D222",
              "nodeType": "function",
              "props": Object {
                "deleteRobot": undefined,
                "inRosters": undefined,
                "robot": Object {
                  "agility": 9,
                  "firstName": "R2",
                  "id": "ZXY987",
                  "lastName": "D2",
                  "speed": 5,
                  "strength": 8,
                  "totalAttrScore": 22,
                },
                "updateName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
